simulators/common/providers/hive/hive.go

```diff
@@ -33,9 +33,11 @@ type host struct {
 var hostProxy *host
 var once sync.Once
 
-// Support this provider type to register it
-func Support() {
-	common.RegisterProvider("hive", GetInstance)
+// New takes the value of the environment variable "HIVE_SIMULATOR"
+// and returns a new common.TestSuiteHost
+func New() (common.TestSuiteHost, error) {
+	config := []byte(fmt.Sprintf(`{"hostURI":%q}`, os.Getenv("HIVE_SIMULATOR")))
+	return GetInstance(config)
```

Comment:
Please add a check which panics if the HIVE_SIMULATOR environment variable is not set.
You can also simplify this by just constructing the HostConfiguration and host objects directly.